### DISCUSSION TITLE
example: system_information - Show TB insted of GB

### DISCUSSION THREAD

**Discussion by ds2k5** [hecrj/iced#1962] [General]
Hello,
I build the binary of the example system_information

https://github.com/iced-rs/iced/tree/master/examples/system_information

But it show TB instead of GB

![mem_top](https://github.com/iced-rs/iced/assets/11292286/bebc8faa-789a-43f2-9ae9-34983fc61df6)

![sysinfo](https://github.com/iced-rs/iced/assets/11292286/9fbf327e-b293-4ba0-9ee5-2cab67c7e967)

My system has 16 GB RAM but the binary show TB
Using: rust 1.70 and Debian 12 (x86_64)

Build with:  `cargo build --release `

Tyred also in a Windows VM
the same issue TB instead of GB

Build with:  `cargo build --release --target x86_64-pc-windows-gnu`

**Comment by lufte** [comment]
It looks like [it's using sysinfo's total_memory](https://github.com/iced-rs/iced/blob/0.9/winit/src/system.rs#L37), which [returns bytes](https://docs.rs/sysinfo/latest/sysinfo/trait.SystemExt.html#tymethod.total_memory), but [iced assumes it's kilobytes](https://github.com/iced-rs/iced/blob/0.9/native/src/system/information.rs#L21).

You can probably write a pull request to fix it.

**Comment by ds2k5** [comment]
@lufte 

thanks - found the line to fix the totalmemory 

```
                let memory_readable =
                    ByteSize::b(information.memory_total).to_string();

```

but the value for used memory is wrong the value is to low 100 insted of 4000 ? the factor is not correct 100 instead of 1000

**Comment by lufte** [comment]
No I think memory_used suffers from the same problem as memory_total, but it refers to the memory used by the Iced app alone, not **all** memory being used. That's maybe why you think it's too low?
